code/modules/projectiles/gun.dm

```diff
@@ -239,6 +243,25 @@
 		user.update_inv_r_hand()
 	feedback_add_details("gun_fired","[src.type]")
 
+/obj/item/weapon/gun/proc/process_laser(atom/target, mob/living/user, params)
+	if(!pointer_icon)
+		return
+
+	var/turf/targloc = get_turf(target)
+
+	//laser pointer image
+	var/list/showto = list()
+	for(var/mob/M in range(7,targloc))
```

Comment:
```view()``` is ```range()``` but with solid (opacity = 1) object checks (p.much the only opaque things we have are walls and Mechs) so that's good for your needs.

However I'd suggest ```viewers()```, which is a list of people who can see the targloc, not a list of things the targloc can see.